src/main/java/com/devcycle/sdk/server/common/model/DevCycleUser.java

```diff
@@ -30,81 +35,165 @@
+
+    @Schema(description = "User's name used to identify the user on the dashboard / target audiences")
+    @JsonProperty("name")
+    private String name;
+
+    @Schema(description = "User's language in ISO 639-1 format")
+    @JsonProperty("language")
+    private String language;
+
+    @Schema(description = "User's country in ISO 3166 alpha-2 format")
+    @JsonProperty("country")
+    private String country;
+
+    @Schema(description = "App Version of the running application")
+    @JsonProperty("appVersion")
+    private String appVersion;
+
+    @Schema(description = "App Build number of the running application")
+    @JsonProperty("appBuild")
+    private String appBuild;
+
+    @Schema(description = "User's custom data to target the user with, data will be logged to DevCycle for use in dashboard.")
+    @JsonProperty("customData")
+    private Map<String, Object> customData;
+
+    @Schema(description = "User's custom data to target the user with, data will not be logged to DevCycle only used for feature bucketing.")
+    @JsonProperty("privateCustomData")
+    private Map<String, Object> privateCustomData;
+
+    @Schema(description = "Date the user was created, Unix epoch timestamp format")
+    @JsonProperty("createdDate")
+    @JsonDeserialize(using = LongTimestampDeserializer.class)
+    @JsonFormat(shape = JsonFormat.Shape.STRING)
+    private Long createdDate;
+
+    @Schema(description = "Date the user was last seen, Unix epoch timestamp format")
+    @JsonProperty("lastSeenDate")
+    @JsonDeserialize(using = LongTimestampDeserializer.class)
+    @JsonFormat(shape = JsonFormat.Shape.STRING)
+    private Long lastSeenDate;
+
+    @Schema(description = "Platform the SDK is running on")
+    @Builder.Default
+    @JsonProperty("platform")
+    private String platform = getPlatformData().getPlatform();
+
+    @Schema(description = "Version of the platform the SDK is running on")
+    @Builder.Default
+    @JsonProperty("platformVersion")
+    private String platformVersion = getPlatformData().getPlatformVersion();
+
+    @Schema(description = "DevCycle SDK type")
+    @Builder.Default
+    @JsonProperty("sdkType")
+    private PlatformData.SdkTypeEnum sdkType = getPlatformData().getSdkType();
+
+    @Schema(description = "DevCycle SDK Version")
+    @Builder.Default
+    @JsonProperty("sdkVersion")
+    private String sdkVersion = getPlatformData().getSdkVersion();
+
+    @Schema(description = "Hostname where the SDK is running")
+    @Builder.Default
+    @JsonProperty("hostname")
+    private String hostname = getPlatformData().getHostname();
+
+    public static PlatformData getPlatformData() {
+        return PlatformData.builder().build();
+    }
+
+    static void setCustomValue(Map<String, Object> customData, String key, Value value) {
+        // Only support boolean, number, and string types for custom data values
+        // ignore all other data
+        if (customData != null && key != null && value != null) {
```

Comment:
If customdata is null - should we not init it as an empty map?

Reply:
Not really, because the map is being passed in. That said this null check is not great as it can potentially cover up a bug in the calling code so I removed it so that a NullPointerException will trigger instead

Reply:
👍🏻